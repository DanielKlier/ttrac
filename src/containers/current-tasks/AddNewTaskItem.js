import * as React from 'react';
import {Button, Clearfix, Glyphicon} from 'react-bootstrap';
import PropTypes from 'prop-types';

class AddNewTaskItem extends React.PureComponent {

    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        onClick: PropTypes.func
    };

    render() {
        return (
            <li className="list-group-item">
                <Clearfix>
                    <span className="add-new-task-text">Add new task</span>
                    <Button className="pull-right" bsStyle="primary"
                            onClick={() => this.props.onClick()}>
                        <Glyphicon glyph="plus"/>
                    </Button>
                </Clearfix>
            </li>
        )
    }
}

export default AddNewTaskItem;
