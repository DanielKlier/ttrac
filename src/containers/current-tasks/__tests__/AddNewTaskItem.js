import * as React from 'react';
import AddNewTaskItem from '../AddNewTaskItem';
import {mount} from 'enzyme';
import toJSON from 'enzyme-to-json';

test('AddNewTaskItem renders correctly', () => {
    const onClick   = jest.fn();
    const component = mount(<AddNewTaskItem onClick={onClick}/>);
    const tree      = toJSON(component);

    expect(tree).toMatchSnapshot();

    component.find('Button').simulate('click');

    expect(onClick).toHaveBeenCalledTimes(1);
});
