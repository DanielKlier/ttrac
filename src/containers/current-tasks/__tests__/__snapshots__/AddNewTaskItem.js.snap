// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewTaskItem renders correctly 1`] = `
<AddNewTaskItem
  onClick={[Function]}
>
  <li
    className="list-group-item"
  >
    <Clearfix
      bsClass="clearfix"
      componentClass="div"
    >
      <div
        className="clearfix"
      >
        <span
          className="add-new-task-text"
        >
          Add new task
        </span>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          className="pull-right"
          disabled={false}
          onClick={[Function]}
        >
          <button
            className="pull-right btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <Glyphicon
              bsClass="glyphicon"
              glyph="plus"
            >
              <span
                className="glyphicon glyphicon-plus"
              />
            </Glyphicon>
          </button>
        </Button>
      </div>
    </Clearfix>
  </li>
</AddNewTaskItem>
`;
